wrapper/src/main/java/software/amazon/jdbc/PluginService.java

```diff
@@ -248,4 +249,12 @@ Connection forceConnect(
   <T> T getPlugin(final Class<T> pluginClazz);
 
   boolean isPluginInUse(final Class<? extends ConnectionPlugin> pluginClazz);
+
+  // JDBC call context functions
+
+  Boolean isPooledConnection();
```

Comment:
If we leave this as `Boolean` instead of `boolean` (see my other comments), can we please add the Nullable annotation here so that the compiler will give a warning if we try to use it as a regular `boolean`?